src/sympc/protocol/beaver/beaver.py

```diff
@@ -98,25 +102,39 @@ def _get_triples(
         )
     elif session.protocol.share_class == ReplicatedSharedTensor:
 
+        if ring_size is None:
+            ring_size = session.ring_size
+        if config is None:
+            config = session.config
+
         a_ptrs: List = []
         b_ptrs: List = []
         for session_ptr in session.session_ptrs:
-            a_ptrs.append(session_ptr.prrs_generate_random_share(a_shape))
-            b_ptrs.append(session_ptr.prrs_generate_random_share(b_shape))
+            a_ptrs.append(
+                session_ptr.prrs_generate_random_share(a_shape, str(ring_size))
+            )
+            b_ptrs.append(
+                session_ptr.prrs_generate_random_share(b_shape, str(ring_size))
+            )
 
         a = MPCTensor(shares=a_ptrs, session=session, shape=a_shape)
         b = MPCTensor(shares=b_ptrs, session=session, shape=b_shape)
-        c = Falcon.mul_semi_honest(a, b, session, op_str, truncate=False, **kwargs)
+        c = Falcon.mul_semi_honest(
+            a, b, session, op_str, ring_size, config, reshare=True, **kwargs
+        )
 
         a_shares = [share.get_copy() for share in a.share_ptrs]
         b_shares = [share.get_copy() for share in b.share_ptrs]
         c_shares = [share.get_copy() for share in c.share_ptrs]
 
-        a_val = sum([a_shares[0].shares[0]] + a_shares[1].shares)
-        b_val = sum([b_shares[0].shares[0]] + b_shares[1].shares)
-        c_val = sum([c_shares[0].shares[0]] + c_shares[1].shares)
+        shares_sum = ReplicatedSharedTensor.shares_sum
+        a_val = shares_sum([a_shares[0].shares[0]] + a_shares[1].shares, ring_size)
+        b_val = shares_sum([b_shares[0].shares[0]] + b_shares[1].shares, ring_size)
+        c_val = shares_sum([c_shares[0].shares[0]] + c_shares[1].shares, ring_size)
+
+        op = ReplicatedSharedTensor.get_op(ring_size, op_str)
```

Comment:
Q: Why do you need the ```get_op```? Can't you directly do?

```
def __add__(self, other):
     if self.ring_size == specific_size:
        # Special add
     else:
        # normal add
```

Reply:
Good ques. For RST addition,sub,mul ,we could use the same ops(`+,-,*`),based on the ring_size ,it would  add.There are scenarios where need to do ops directly on torch.tensor(beaver triples, triple verification,reconstruction...). I initially thought to override torch method,but as you suggested in slack, I thought not to do it.I kept the ops retrieval for  tensors in a central location(i.e RSTensor) ,so we could dynamically add ops on requirement.
https://github.com/OpenMined/SyMPC/blob/113895e1e15b253b2e7467a05cebf0f2876e139d/src/sympc/tensor/replicatedshare_tensor.py#L240